### PR TITLE
Add more validation mechanism to text control

### PR DESCRIPTION
**Pull request by jess0407**
https://rockabox.tpondemand.com/entity/11855
- monetary value cannot be negative
- add max value attribute and validation
- add min value attribute and validation

test and merge https://github.com/rockabox/rbx_campaign_management/pull/343

### CODE REVIEW

File: src/rb-text-control/rb-text-control-directive.js [hunk 1]
```diff
@@ -50,7 +50,8 @@ define([
                 small: '@',
                 onChange: '&',
                 modelOptions: '@',
-                maxDigits: '@'
+                maxDigits: '@',
```
Comment (gregbenner) [inline src/rb-text-control/rb-text-control-directive.js]: Why not use built in 'max' https://docs.angularjs.org/api/ng/input/input%5Bnumber%5D ?

Comment (jess0407) [inline src/rb-text-control/rb-text-control-directive.js]: @gregbenner this pr is still under construction, purely to test on CM, and can be reviewed later.

Comment (gregbenner) [inline src/rb-text-control/rb-text-control-directive.js]: Ok ignore it till you're ready :)

Comment (jess0407) [inline src/rb-text-control/rb-text-control-directive.js]: Note the context of this component is rb-text-control, not native input, therefore build in min and max does not apply here.

Comment (gregbenner) [inline src/rb-text-control/rb-text-control-directive.js]: I would just transclude the attribute onto the input and let the built in validator do the work.
Could still rename as you have, but let the native angular validation do the work, unless you need speciall functionality?

Comment (jess0407) [inline src/rb-text-control/rb-text-control-directive.js]: Yep, the current implementation require less work, and it works. 
Note that angular produce validation attributes as something like myForm.input.$valid
But we need something like form[name].$error.exceedMaxValue and
form[name].$error.exceedMinValue to show and hide the relevant validation error message.
Can't see why should we reimplement this (it's not gonna improve much if any performance, and it's going to be more complicated and require more redundant code).

File: src/rb-text-control/custom-validation-directive.js [hunk 2]
```diff
@@ -44,6 +45,26 @@ define([
...
+                        var result;
+                        if (!scope.maxValue) {
+                            return true;
+                        } else {
+                            result = (modelValue <= Number(scope.maxValue, 10));
+                        }
+                        return ctrl.$isEmpty(modelValue) || result;
+                    };
+
+                    ctrl.$validators.exceedMinValue = function (modelValue, viewValue) {
```
Comment (gregbenner) [inline src/rb-text-control/custom-validation-directive.js]: exceed min value?

Comment (jess0407) [inline src/rb-text-control/custom-validation-directive.js]: Apologies maybe my english is not that good.
But I believe 'exceed' make sense here just so as here below: 
http://www.programmingincpp.com/exceed-integer-limits.html
I am open to suggestions though, what is the a good word here to a native speaker?

Comment (gregbenner) [inline src/rb-text-control/custom-validation-directive.js]: No worries. exceed = be greater in number or size than (a quantity, number, or other measurable thing). "production costs have exceeded £60,000"
Would just keep it simple like:  overMax / underMin

File: src/rb-text-control/rb-text-control-directive.js (hunk 1 above)
Comment (gregbenner) [inline src/rb-text-control/rb-text-control-directive.js]: I would write the validations a bit simpler if it makes sense to you
<code>
if (scope.maxValue) {
	ctrl.$validators.exceedMaxValue = function (modelValue) {
		var result = (modelValue <= Number(scope.maxValue, 10)) || false;
		return result;
	}
} </code>
This way also the validator isn't attached and evaluated unless the scope property exists.

File: src/rb-text-control/custom-validation-directive.js (hunk 2 above)
Comment (jess0407) [inline src/rb-text-control/custom-validation-directive.js]: Have updated using underMinValue and overMaxValue.

File: src/rb-text-control/rb-text-control-directive.js (hunk 1 above)
Comment (jess0407) [inline src/rb-text-control/rb-text-control-directive.js]: This is actually a good point, have refactored so as below:
screen shot 2015-10-05 at 17 45 13

**Pull request merged**